web/src/message_events.js

```diff
@@ -542,8 +540,15 @@ export function update_messages(events) {
         // edit.  Doing so could save significant work, since most
         // topic edits will not match the current topic narrow in
         // large organizations.
-        if (!changed_narrow && message_lists.current?.narrowed) {
-            message_lists.current.update_muting_and_rerender();
+
+        for (const list of message_lists.all_rendered_message_lists()) {
+            if (changed_narrow && list === message_lists.current && !list.preserve_rendered_state) {
```

Comment:
Do we need the `preserve_rendered_state` check here?

If we switched to a new narrow already, then `message_lists.current` must be that narrow -- and we know a rerender already occurred, so it seems like it shouldn't be relevant whether `preserve_rendered_state` is set -- the point is we'd just be duplicating work.

Reply:
Oh, also, just noticed that the if/else clauses here could potentially be combined? But let's leave that for a follow-up PR, this has enough going on, and probably that follow-up PR might also want to address the `BUG` and `TODO` comments in this part of the code.

Reply:
> Do we need the preserve_rendered_state check here?

Nice catch. Removed.